server/test/durablejobs/jobs/ConvertAddressServiceAreaToArrayJobTest.java

```diff
@@ -0,0 +1,369 @@
+    ApplicationModel application = resourceCreator.insertActiveApplication(applicant, program);
+    application.setApplicantData(applicant.getApplicantData());
+    application.save();
+
+    // act
+    runJob();
+
+    // assert
+    ApplicationModel newApplication =
+        applicationRepository.getApplication(application.id).toCompletableFuture().join().get();
+    ApplicantData newApplicationData = newApplication.getApplicantData();
+    assertServiceAreaValues(Path.create("applicant.applicant_address"), newApplicationData);
+    assertServiceAreaValues(
+        Path.create("applicant.enumerator_one[0].applicant_address_child"), newApplicationData);
+    assertServiceAreaValues(
+        Path.create("applicant.enumerator_one[1].applicant_address_child"), newApplicationData);
+  }
+
+  private static void runJob() {
+    ConvertAddressServiceAreaToArrayJob job =
+        new ConvertAddressServiceAreaToArrayJob(
+            new PersistedDurableJobModel("fake-job", JobType.RUN_ONCE, Instant.now()));
+
+    job.run();
+  }
+
+  private static void assertServiceAreaValues(Path rootPath, ApplicantData newApplicationData) {
+    // Assert top service area keys are as expected
+    assertThat(newApplicationData.hasPath(rootPath.join(Scalar.SERVICE_AREA))).isFalse();
+    assertThat(newApplicationData.hasPath(rootPath.join(Scalar.SERVICE_AREAS))).isTrue();
+
+    // Assert expected child keys exist
+    Path serviceAreaPath = rootPath.join(Scalar.SERVICE_AREAS).asArrayElement();
+    assertThat(
+            newApplicationData
+                .readString(serviceAreaPath.atIndex(0).join(Scalar.SERVICE_AREA_ID))
+                .isPresent())
+        .isTrue();
+    assertThat(
+            newApplicationData
+                .readString(serviceAreaPath.atIndex(0).join(Scalar.SERVICE_AREA_STATE))
+                .isPresent())
+        .isTrue();
+    assertThat(
+            newApplicationData
+                .readString(serviceAreaPath.atIndex(0).join(Scalar.TIMESTAMP))
+                .isPresent())
+        .isTrue();
+    assertThat(
+            newApplicationData
+                .readString(serviceAreaPath.atIndex(1).join(Scalar.SERVICE_AREA_ID))
+                .isPresent())
+        .isTrue();
+    assertThat(
+            newApplicationData
+                .readString(serviceAreaPath.atIndex(1).join(Scalar.SERVICE_AREA_STATE))
+                .isPresent())
+        .isTrue();
+    assertThat(
+            newApplicationData
+                .readString(serviceAreaPath.atIndex(1).join(Scalar.TIMESTAMP))
+                .isPresent())
+        .isTrue();
+
+    // Assert getting the typed list returns expected values
+    var serviceAreas = newApplicationData.readServiceAreaList(serviceAreaPath);
+    assertThat(serviceAreas.isPresent()).isTrue();
+    assertThat(serviceAreas.get().size()).isEqualTo(2);
+
+    var bloomington =
+        serviceAreas.get().stream()
+            .filter(x -> x.getServiceAreaId().equals("Bloomington"))
```

Comment:
**thought, non-blocking**: in general I don't _love_ having the expected values outside of the test method, since then you have to go look somewhere else to see everything that the test is looking for, but I don't feel strongly enough in this case to ask you to refactor it, esp since this is a one-off thing.

Reply:
Yeah, I get where you're coming from. In this case I'm more concerned with being certain everything is uniform.